draft-ietf-masque-connect-ip-dns.md

```diff
@@ -348,6 +359,75 @@ DNS Configuration = {
 ~~~
 {: #ex-split title="Split Tunnel Example"}
 
+# PREF64 Configuration Mechanism
+
+This document defines a new capsule type, `PREF64`, that allows a CONNECT-IP peer
+to communicate the IPv6 NAT64 prefixes to be used for IPv6/IPv4 address synthesis.
+This information enables an endpoint operating in an IPv6-only environment to
+construct IPv4-reachable addresses from IPv6 literals when a NAT64 translator is in use.
+
+## PREF64 Capsule
+
+Each PREF64 capsule conveys zero or more NAT64 prefixes. If multiple capsules are sent
+in the same direction, the most recent one replaces any previously advertised prefixes.
+Empty PREF64 capsule is used to inform that NAT64 prefixes are not available.
+
+The capsule has the following structure (see {{iana}} for the value of the capsule type):
+
+~~~
+PREF64 Capsule {
+  Type (i) = PREF64,
+  Length (i),
+  NAT64 Prefix (104) ...,
+}
+~~~
+{: #pref64-format title="PREF64 Capsule Format"}
+
+Individual NAT64 prefix has the following structure
+
+~~~
+NAT64 Prefix {
+  Prefix Length (8),
+  Prefix (96),
+}
+~~~
+{: #nat64-prefix title="NAT64 Prefix Format"}
+
+NAT64 prefix fields are:
+
+Prefix Length:
+
+: The length of the NAT64 prefix in bits encoded as a single byte. Valid values are 32, 40, 48,
+56, 64, and 96. These lengths correspond to the prefix sizes permitted for the IPv6-to-IPv4
+address synthesis algorithm described in {{Section 2.2 of !IPv6-TRANSLATOR=RFC6052}}.
+
+Prefix:
+
+: The highest 96 bits of the IPv6 prefix, encoded in network byte order.
+
+## Handling
```

Comment:
Can you add a sentence like "If an endpoint receives a capsule that does not meet one of the requirements above, or with a length that is not a multiple of 13 bytes, it MUST treat it as malformed. Note that an empty PREF64 capsule is valid; it indicates that no NAT64 prefixes are available on this network."